src/Datadog.Sketches/Stores/Bin.cs

```diff
@@ -0,0 +1,59 @@
+// <copyright file="Bin.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2022 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.Diagnostics;
+
+namespace Datadog.Sketches.Stores
+{
+    /// <summary>
+    /// A pair of index and count.
+    /// </summary>
+    [DebuggerDisplay("({Index}, {Count})")]
+    public readonly struct Bin : IEquatable<Bin>
```

Comment:
Optional, but I think you could _almost_ reduce this whole file to:

```csharp
namespace Datadog.Sketches.Stores;
readonly record struct Bin(int Index, double Count);
```

It's not quite the same, as it creates `Index` and `Count` properties rather than fields. Just sayin'.

Reply:
I don't think using records with net461 is recommended?

Reply:
Oh really, I hadn't heard that? [AFAIK, it doesn't lower to anything fancy](https://sharplab.io/#v2:EYLgtghglgdgPgJwKYQCYHsYBsCeACZAY3QVTwGcAXBAV0MrwCFYAKWBgSRlSQA8AaPBhrAsSPAGF0NGJQCUAbiA).